packages/console/src/hooks/use-tenants.ts

```diff
@@ -0,0 +1,86 @@
+import { useLogto } from '@logto/react';
+import { type TenantInfo } from '@logto/schemas';
+import { type Optional, trySafe } from '@silverhand/essentials';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+
+import useSwrFetcher from './use-swr-fetcher';
+
+const useTenants = () => {
+  const cloudApi = useCloudApi();
+  const { signIn, getAccessToken } = useLogto();
+  const {
+    tenants,
+    setTenants: originalSetTenants,
+    currentTenantId,
+    setCurrentTenantId,
+    setIsSettle,
+    navigate,
+  } = useContext(TenantsContext);
+  const [error, setError] = useState<Error>();
+
+  const fetcher = useSwrFetcher<TenantInfo[]>(cloudApi);
+  const {
+    data: availableTenants,
+    error: requestError,
+    mutate,
+  } = useSWR<TenantInfo[], Error>('/api/tenants', fetcher);
+  const isLoading = Boolean(!availableTenants && !requestError);
+  const isLoaded = Boolean(availableTenants && !requestError);
+
+  const setTenants = useCallback(
+    (tenants: TenantInfo[]) => {
+      originalSetTenants(tenants);
+      void mutate(tenants);
+    },
+    [mutate, originalSetTenants]
+  );
```

Comment:
Again, since we don't need to keep `tenants` in context, we can remove this method, calling only `mutate` when the update is done